language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java

```diff
@@ -91,12 +92,12 @@ public List<PackageInfo> getLocalRepoPackages(PackageRepository repository) {
      *
      * @return {@link List} of remote repo packages
      */
-    public List<PackageInfo> getRemoteRepoPackages(PackageRepository repository) {
+    public List<ModuleInfo> getRemoteRepoPackages(PackageRepository repository) {
         if (!this.remoteRepoPackages.isEmpty()) {
             return this.remoteRepoPackages;
         }
         this.remoteRepoPackages.addAll(checkAndResolvePackagesFromRepository(repository, Collections.emptyList(),
-                Collections.emptySet()));
+                this.remoteRepoPackages.stream().map(ModuleInfo::packageIdentifier).collect(Collectors.toSet())));
```

Comment:
Here, the `remoteRepoPackages` is empty (checked in `if` condition above). Therefore, no point in converting this to a set. Just pass an empty list as it was earlier.